Broken Build

When trying to clone and build from default branch or from  'v2.4.1' tag using:

```bash
# Download the source code
git clone https://github.com/ProvableHQ/leo
# or 'git clone -b v2.4.1 --recurse-submodules --single-branch https://github.com/ProvableHQ/leo'
cd leo

# Install 'leo'
cargo install --path .
```

It results in the following error:

```bash
error: failed to compile `leo-lang v2.4.1 (<REDACTED>/leo)`, intermediate artifacts can be found at `<REDACTED>/leo/target`.
To reuse those artifacts with a future compilation, set the environment variable `CARGO_TARGET_DIR` to that path.

Caused by:
  rustc 1.82.0 is not supported by the following package:
    snarkvm@1.4.0 requires rustc 1.83.0
  Either upgrade rustc or select compatible dependency versions with
  `cargo update <name>@<current-ver> --precise <compatible-ver>`
  where `<compatible-ver>` is the latest version supporting rustc 1.82.0
```

which I can easily workaround by forcing rustc 1.83.0 using `cargo +1.83.0 install --path .`
Then I get the following errors:
```bash
error[E0023]: this pattern has 3 fields, but the corresponding tuple variant has 4 fields
   --> leo/cli/commands/execute.rs:204:34
    |
204 |             Transaction::Execute(_, execution, _) => execution,
    |                                  ^  ^^^^^^^^^  ^ expected 4 fields, found 3
    |
   ::: <REDACTED>/.cargo/registry/src/index.crates.io-6f17d22bba15001f/snarkvm-ledger-block-1.4.0/src/transaction/mod.rs:55:13
    |
55  |     Execute(N::TransactionID, ExecutionID<N>, Box<Execution<N>>, Option<Fee<N>>),
    |             ----------------  --------------  -----------------  -------------- tuple variant has 4 fields
    |
help: use `_` to explicitly ignore each field
    |
204 |             Transaction::Execute(_, execution, _, _) => execution,
    |                                                 +++

error[E0599]: no associated item named `CONSENSUS_V2_HEIGHT` found for associated type `<A as snarkvm::snarkvm_circuit::Environment>::Network` in the current scope
   --> leo/cli/commands/execute.rs:221:49
    |
221 |                         if height < A::Network::CONSENSUS_V2_HEIGHT {
    |                                                 ^^^^^^^^^^^^^^^^^^^ associated item not found in `<A as Environment>::Network`
    |
help: there is an associated function `CONSENSUS_HEIGHT` with a similar name
   --> <REDACTED>/.cargo/registry/src/index.crates.io-6f17d22bba15001f/snarkvm-console-network-1.4.0/src/lib.rs:253:5
    |
253 |     fn CONSENSUS_HEIGHT(version: ConsensusVersion) -> Result<u32> {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Some errors have detailed explanations: E0023, E0599.
For more information about an error, try `rustc --explain E0023`.
error: could not compile `leo-lang` (lib) due to 2 previous errors
warning: build failed, waiting for other jobs to finish...
error: failed to compile `leo-lang v2.4.1 (<REDACTED>/leo)`, intermediate artifacts can be found at `<REDACTED>/leo/target`.
To reuse those artifacts with a future compilation, set the environment variable `CARGO_TARGET_DIR` to that path.
```

This build based on v2.4.1 tag has worked a few days ago and I noticed that since then, version '1.4.0' of `snarkvm` and related dependencies has been published to crates.io so perhaps some of the dependencies automatically resolve the newer versions hence the errors.

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -111,7 +111,7 @@ default-features = false
 version = "1.11.1"
 
 [workspace.dependencies.snarkvm]
-version = "1.3.0"
+version = "=1.3.0"
 
 [workspace.dependencies.serde]
 version = "1.0.214"

